Dispatcher.php

```diff
@@ -102,15 +114,32 @@ public function send(Message $message, $templateName = '', $templateContent = ar
 
         if (!empty($templateName)) {
             return $this->service->messages->sendTemplate($templateName, $templateContent, $message->toArray(), $async, $ipPool, $sendAt);
-
         }
 
         return $this->service->messages->send($message->toArray(), $async, $ipPool, $sendAt);
-
     }
 
+    private function useProxy()
+    {
+        return $this->proxy['use'];
+    }
 
-
-
-
+    private function addCurlProxyOptions()
+    {
+        if ($this->proxy['host'] !== null) {
+            curl_setopt($this->service->ch, CURLOPT_PROXY, $this->proxy['host']);
+        }
+        
+        if ($this->proxy['port'] !== null) {
+            curl_setopt($this->service->ch, CURLOPT_PROXYPORT, $this->proxy['port']);
+        }
+        
+        if ($this->proxy['user'] !== null && $this->proxy['password'] !== null) {
```

Comment:
> seems perfectly legit AFAIK.

Do you have any publications ? I have made some researches and I found nothing.

Anyway, it is possible to have this behavior with the following configuration:
```yaml
proxy:
    #...
    user: username 
    password: "" # empty string
```

But it is strongly not recommended to have an empty password ;)